scripts/survival_migration/survivalDataMigration.py

```diff
@@ -0,0 +1,196 @@
+    elif value.lower() in vocabulariesMappingToZero:
+        return "0:" + value
+    else:
+        sys.exit("cannot find the mapping for vocabulary {} , please provide the the mapping rules with option -a".format(value))
+        return value
+
+def generateVocabularies(vocabulariesMappingToOne, vocabulariesMappingToZero, vocabulariesFile):
+    df = pd.read_csv(vocabulariesFile, sep='\t')
+    oneValues = df[MAPPING_TO_ONE_COLUMN].tolist()
+    for unparsedValue in oneValues:
+        for parsedValue in unparsedValue.split(","):
+            vocabulariesMappingToOne.append(parsedValue)
+    zeroValues = df[MAPPING_TO_ZERO_COLUMN].tolist()
+    for unparsedValue in zeroValues:
+        for parsedValue in unparsedValue.split(","):
+            vocabulariesMappingToZero.append(parsedValue)
+
+def splitAdditionalVocabularies(voc, oneVoc, zeroVoc, vocabulariesFile):
+    zeroAndOneSplit = voc.split('#')
+    newVocAddingToZero = []
+    newVocAddingToOne = []
+    # just defined one or zero
+    for splitString in zeroAndOneSplit:
+        mapAndValuesSplit = splitString.split(':')
+        if (len(mapAndValuesSplit) == 2):
+            mapId = mapAndValuesSplit[0]
+            values = mapAndValuesSplit[1].split(',')
+            if mapId == '0':
+                for value in values:
+                    if value.lower() not in zeroVoc:
+                        zeroVoc.append(value.lower())
+                        newVocAddingToZero.append(value.lower())
+            elif mapId == '1':
+                for value in values:
+                    if value.lower() not in oneVoc:
+                        oneVoc.append(value.lower())
+                        newVocAddingToOne.append(value.lower())
+        else:
+            print("additionalVocabularies format is incorrect, please check the example by using -h option")
+
+    # save vocabularies to file
+    separator = ','
+    oneUpdateString = separator.join(newVocAddingToOne)
+    zeroUpdateString = separator.join(newVocAddingToZero)
+    df = pd.read_csv(vocabulariesFile, sep='\t')
+
+    if len(newVocAddingToZero) > 0 or len(newVocAddingToOne) > 0:
+        # get vocabularies
+        df2 = pd.DataFrame([[oneUpdateString, zeroUpdateString]], columns=[MAPPING_TO_ONE_COLUMN, MAPPING_TO_ZERO_COLUMN])
+        df = df.append(df2, ignore_index=True)
+        if len(newVocAddingToOne) > 0:
+            print("vocabulary {} has been added to one mapping".format(oneUpdateString))
+        if len(newVocAddingToZero) > 0:
+            print("vocabulary {} has been added to zero mapping".format(zeroUpdateString))
+
+    df.to_csv(vocabulariesFile, index=False, sep='\t', header=True)
+
+def migrate_file(args):
+    clinicalFile = args.clinicalFile
+    vocabulariesFile = args.vocabulariesFile
+    shouldOverrive = args.override
+    newFile = os.path.dirname(args.clinicalFile) + "/new_data_clinical_patients.txt"
+    tmpFile = os.path.dirname(args.clinicalFile) + "/tmp_data_clinical_patients.txt"
```

Comment:
Should the out and tmp file follow the same naming pattern as input file with a suffix of migrated or something like that? Not sure if everyone uses `data_clinical_patients.txt`